docs/guides/example_messaging.md

```diff
@@ -0,0 +1,331 @@
+        "units": "Degrees-Celsius",
+      },
+      "return_air_temperature_sensor": {
+        "units": "Degrees-Celsius",
+      },
+      "fan_speed_percentage_command": {
+        "units": "Percent",
+      },
+      "zone_air_temperature_setpoint": {
+        "writeable": true,
+        "units": "Degrees-Celsius",
+      }
+    }
+  }
+}
+```
+
+Additional fields, such as baseline values and testing targets omitted as well as optional
+descriptive fields. It is suggested these are filled in where available.
+
+## Message Flow
+
+![sequence diagram message flow](images/example-sequence-diagram.png)
+
+### Initialization 
+
+#### (1) Initial Configuration
+
+Using the registrar tool, a configuration message is sent to the device based on metadata file
+
+```json
+{
+  "version": 1,
+  "timestamp": "2021-12-21T13:00:00.131Z",
+  "system": {
+    "min_loglevel": 600
+  },
+  "pointset": {
+    "sample_limit_sec": 15,
+    "sample_rate_sec": 60,
+    "points": {
+      "cooling_valve_percentage_command": {
+      },
+      "supply_air_temperature_sensor": {
+      },
+      "return_air_temperature_sensor": {
+      },
+      "fan_speed_percentage_command": {
+      },
+      "zone_air_temperature_setpoint": {
+      }
+    }
+  }
+}
+```
+
+#### (2) State
+
+After receiving a config message, the device sends a [state](../messages/state.md) message as
+defined by [config and state sequence](../specs/sequences/config.md)
+
+```json
+{
+  "version": 1,
+  "timestamp": "2021-12-21T13:00:01.215Z",
+  "system": {
+    "make_model": "ACME FCU Controller", 
+    "firmware": {
+      "version": "3.2a"
+    },
+    "serial_no": "182732142",
+    "last_config": "2021-12-21T13:00:00.131Z", // Matches the timestamp of the last config message
+    "operational": true
+    },
+  "pointset": {
+    "points": {
+      "cooling_valve_percentage_command": {
+      },
+      "supply_air_temperature_sensor": {
+      },
+      "return_air_temperature_sensor": {
+      },
+      "fan_speed_percentage_command": {
+      },
+      "zone_air_temperature_setpoint": {
+      }
+    }
+  }
+}
+```
+
+### Operation
+
+#### (3) Telemetry (Pointset Event)
+
+The telemetry will send data at an interval according to the `sample_limit_sec` and
+`sample_rate_sec` from the [config](config.md) message.
+
+In this example with the following parameters:
+* `sample_rate_sec`: 15
+* `sample_limit_sec`: 60
+The telemetry will be nominally sent every 60 seconds
+
+```json
+{
+  "version": 1,
+  "timestamp": "2021-12-21T13:00:02.053Z",
+  "points": {
+    "cooling_valve_percentage_command": {
+        "present_value": 50
+    },
+    "supply_air_temperature_sensor": {
+        "present_value": 22.3
+    },
+    "return_air_temperature_sensor": {
+        "present_value": 26.1
+    },
+    "fan_speed_percentage_command": {
+        "present_value": 50
+    },
+    "zone_air_temperature_setpoint": {
+        "present_value": 26.0
+    }
+  }
+}
+```
```

Comment:
I think specific examples for any examples or tutorials is needed rather than the generic examples already in the tests folder. 

Can we create examples in the test folder e.g. `tests/state.tests/examples_messaging_s1.json` and link to them from the examples?

Reply:
Couldn’t agree with @noursaidi more here. I just learned far more from skimming this example in a couple mins than I have staring at the documentation for several days. The piecemeal documentation with 10 different links doesn’t work for someone coming into this and that’s who this is for. If there’s an issue with things getting out of date then I would suggest stating whatever version this was written in.

Reply:
Yes -- you can put as many examples of whatever you want in the tests/ folder. The advantage of having them there is that they will be validated directly to make sure nothing breaks.  Also, I'm enabling comments in the JSON parser for the tools, so that you can now add explanatory comments directly to the example files (see https://github.com/grafnu/udmi/blob/enumeration/tests/event_discovery.tests/enumeration.json). It'll be enabled once I get that PR in, or it's a simple two-line code change (see https://github.com/faucetsdn/udmi/pull/235/files file validator/src/main/java/com/google/daq/mqtt/validator/Validator.java)